### DISCUSSION TITLE
What is `--flashattention` ?

### DISCUSSION THREAD

**Discussion by VL4DST3R** [LostRuins/koboldcpp#890] [Q&A] [answered]
I've seen it mentioned in the last [release](https://github.com/LostRuins/koboldcpp/releases/tag/v1.67) but there is no mention of it on the wiki nor I could find anything relevant within issues.

**Comment by LostRuins** [comment]
It's a more efficient algorithm that is generally faster and less memory intensive

**Reply by VL4DST3R** [reply]
So... should we enable it on every gguf model? Any downsides?

**Reply by LostRuins** [reply]
Poor performance on older cards

**Reply by sardior** [reply]
What counts as an "older card" :laughing:? I added this flag for a quick test, and on my AMD Radeon RX 7900 XTX the token generation speed tanked by at least 75%.

**Reply by VL4DST3R** [reply, marked as answer]
I guess yours counts as an "older card" then lol. Jokes aside I noticed that amd cards are usually hit or miss when it comes to the latest optimizations.

**Reply by cmdrscotty** [reply]
In my experience, it tends to come with "was this implemented in Vulkan?" 

Running koboldcpp on windows generally comes down to preferring Vulkan over trying to get rocm working (in my experience)

**Reply by LostRuins** [reply]
No, flash attn does not work well on vulkan.
